tests/v3_api/common.py

```diff
@@ -391,22 +406,72 @@ def validate_ingress_using_endpoint(p_client, ingress, workloads,
                     public_endpoint["hostname"]
                 if "path" in public_endpoint.keys():
                     url += public_endpoint["path"]
-    time.sleep(5)
+
     target_hit_list = target_name_list[:]
-    for i in range(1, 20):
+
+    # Wait for Ingress link to be functional
+    check_if_curl_link_is_functional(url, waitforingresslinktobefunctional)
+
+    for i in range(1, 50):
         if len(target_hit_list) == 0:
             break
         cmd = "curl " + url
         print cmd
         result = run_command(cmd)
         result = result.rstrip()
         print result
-        assert result in target_name_list
         if result in target_hit_list:
+            assert result in target_name_list
             target_hit_list.remove(result)
+    print "Length of Target hit list"
+    print len(target_hit_list)
     assert len(target_hit_list) == 0
 
 
+def check_if_endpoint_is_available(client, ingress, timeout):
+    start = time.time()
+    ingressdata = []
+    print ingressdata
+    # Wait for ingress publicendpoints to get populated
+    while not ingressdata:
+        if time.time() - start > timeout:
+            exceptionMsg = 'Timeout waiting for ingress publicendpoints ' \
+                           'to get populated'
+            raise Exception(exceptionMsg)
+        time.sleep(.5)
+        client.reload(ingress)
+        key = "publicEndpoints"
+        ingress = client.reload(ingress)
+        if key in ingress:
+            ingressdata = ingress['publicEndpoints'][0]
+        else:
+            ingressdata = []
+    print ingressdata
+
+
+def check_if_curl_link_is_functional(url, waitforingresslinktobefunctional):
+    curlresult = False
+    start = time.time()
+    while not curlresult:
+        currenttime = time.time()
+        delta = currenttime - start
+        if delta > waitforingresslinktobefunctional:
+            exceptionMsg = 'Timeout waiting for ingress link to work'
+            raise Exception(exceptionMsg)
+        time.sleep(1)
+        cmd = "curl -I " + url
+        result = run_command(cmd)
+        result = result.rstrip()
```

Comment:
It should be sufficient to wait for "200" or wait until there is no connection error ?

Reply:
GKE gives intermediate/temporary errors. Hence the check for error is required

Reply:
Removed the check for errors